mlir/include/mlir/Dialect/OpenMP/OpenMPOpsInterfaces.td

```diff
@@ -69,6 +69,78 @@ def ReductionClauseInterface : OpInterface<"ReductionClauseInterface"> {
   ];
 }
 
+def LoopWrapperInterface : OpInterface<"LoopWrapperInterface"> {
+  let description = [{
+    OpenMP operations that can wrap a single canonical loop. When taking a
+    wrapper role, these operations must only contain a single region with a
+    single block in which there's a single operation and a terminator. That
+    nested operation must be another loop wrapper or an `omp.canonical_loop`.
+  }];
+
+  let cppNamespace = "::mlir::omp";
+
+  let methods = [
+    InterfaceMethod<
+      /*description=*/[{
+        Tell whether the operation is taking the role of a loop wrapper. That
+        is, it has a single region with a single block in which there are two
+        operations: another wrapper or canonical loop and a terminator.
+      }],
+      /*retTy=*/"bool",
+      /*methodName=*/"isWrapper",
+      (ins ), [{}], [{
+        if ($_op->getNumRegions() != 1)
+          return false;
+
+        Region &r = $_op->getRegion(0);
+        if (!r.hasOneBlock())
+          return false;
+
+        llvm::SmallVector<::mlir::Operation *, 2> ops(r.getOps());
+        if (ops.size() != 2)
+          return false;
+
+        return isa<::mlir::omp::CanonicalLoopOp,
+                   ::mlir::omp::LoopWrapperInterface>(ops.front()) &&
+               ops.back()->hasTrait<::mlir::OpTrait::IsTerminator>();
+      }]
+    >,
+    InterfaceMethod<
+      /*description=*/[{
+        If this is a loop wrapper and there is another wrapper immediately
+        nested inside, return that operation.
+      }],
+      /*retTy=*/"::mlir::Operation *",
```

Comment:
nit: I think here we can also be more specific about the returned type: `LoopWrapperInterface`.

Reply:
Done.